src/main/java/andelu/Storage.java

```diff
@@ -0,0 +1,211 @@
+     * @param isOverwrite True for overwrite the previous data. False for appending the data.
+     */
+    public void writeArrayListToFile(ArrayList<Task> tasks, boolean isOverwrite) {
+        try {
+            if (isOverwrite) {
+                Files.write(filePath,
+                        convertTasksToString(tasks).getBytes(),
+                        StandardOpenOption.CREATE,
+                        StandardOpenOption.WRITE,
+                        StandardOpenOption.TRUNCATE_EXISTING);
+            } else {
+                Files.write(filePath,
+                        convertTasksToString(tasks).getBytes(),
+                        StandardOpenOption.CREATE,
+                        StandardOpenOption.APPEND);
+            }
+        } catch (IOException io) {
+            Ui ui = new Ui();
+            ui.printAnyStatement("There is an error when writing to file. The error is " + io.getMessage());
+        }
+    }
+
+    /**
+     * Loads Tasks from the file.
+     *
+     * @return an ArrayList of Tasks.
+     */
+    public ArrayList<Task> loadTasksFromFile() throws AndeluException {
+        ArrayList<Task> result = new ArrayList<>();
+
+        try {
+            List<String> fileContentLines = Files.readAllLines(filePath);
+            result = convertStringListToTasks(fileContentLines);
+        } catch (IOException io) {
+            throw new AndeluException("There is an error when reading the file.");
+        }
+
+        return result;
+    }
+
+    /**
+     * Converts an ArrayList of Tasks to String.
+     *
+     * @param tasks An arrayList of Tasks.
+     * @return String format of the list of Tasks.
+     */
+    private String convertTasksToString(ArrayList<Task> tasks) {
+        StringBuilder result = new StringBuilder();
+
+        for (Task task: tasks) {
+            if (task instanceof ToDo) {
+                result.append("T | ");
+                result.append(task.getStatusIcon().equals("X") ? "1 | " : "0 | ");
+                result.append(task.getDescription());
+                result.append(System.getProperty("line.separator"));
+            } else if (task instanceof Deadline) {
+                result.append("D | ");
+                result.append(task.getStatusIcon().equals("X") ? "1 | " : "0 | ");
+                result.append(task.getDescription() + " | ");
+                result.append(((Deadline) task).getBy());
+                result.append(System.getProperty("line.separator"));
+            } else {
+                result.append("E | ");
+                result.append(task.getStatusIcon().equals("X") ? "1 | " : "0 | ");
+                result.append(task.getDescription() + " | ");
+                result.append(((Event) task).getStart() + " | ");
+                result.append(((Event) task).getEnd());
+                result.append(System.getProperty("line.separator"));
+            }
+
+        }
+
+        return result.toString();
+    }
```

Comment:
### #General
I really like how you format your code. It is very clean and readable!